data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java

```diff
@@ -502,6 +511,51 @@ void aggregateWithHistogramAggregateAction() throws InterruptedException, NoSuch
         }
     }
 
+    @ParameterizedTest
+    @ValueSource(doubles = {20.0, 40.0, 60.0})
+    void aggregateWithTailSamplerAction(final double testPercent) throws InterruptedException, NoSuchFieldException, IllegalAccessException {
+        final Duration testWaitPeriod = Duration.ofSeconds(3);
+        final String testErrorCondition = "/status == "+ERROR_STATUS;
+        when(tailSamplerAggregateActionConfig.getPercent()).thenReturn(testPercent);
+        when(tailSamplerAggregateActionConfig.getWaitPeriod()).thenReturn(testWaitPeriod);
+        when(tailSamplerAggregateActionConfig.getErrorCondition()).thenReturn(testErrorCondition);
+        doAnswer(a -> {
+            Event event = (Event)a.getArgument(1);
+            return event.get("status", Integer.class) == ERROR_STATUS;
+        }).when(expressionEvaluator).evaluateConditional(eq(testErrorCondition), any(Event.class));
+        aggregateAction = new TailSamplerAggregateAction(tailSamplerAggregateActionConfig, expressionEvaluator);
+        when(pluginFactory.loadPlugin(eq(AggregateAction.class), any(PluginSetting.class))).thenReturn(aggregateAction);
+        when(aggregateProcessorConfig.getGroupDuration()).thenReturn(Duration.ofSeconds(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE));
+        when(aggregateProcessorConfig.getIdentificationKeys()).thenReturn(List.of("traceId"));
+        final AggregateProcessor objectUnderTest = createObjectUnderTest();
+        final ExecutorService executorService = Executors.newFixedThreadPool(NUM_THREADS);
+
+        final int numberOfErrorTraces = 2;
+        final int numberOfSpans = 5;
+        eventBatch = getBatchOfEventsForTailSampling(numberOfErrorTraces, numberOfSpans);
+        objectUnderTest.doExecute(eventBatch);
+        Thread.sleep(GROUP_DURATION_FOR_ONLY_SINGLE_CONCLUDE * 1000);
+
+        for (int i = 0; i < NUM_THREADS; i++) {
+            executorService.execute(() -> {
+                final List<Record<Event>> recordsOut = (List<Record<Event>>) objectUnderTest.doExecute(eventBatch);
+                assertThat(recordsOut.size(), equalTo(0));
```

Comment:
I don't think this assertion is going to work as expected. This is running in another thread, so an error here will not fail the test.

Perhaps follow this pattern:

https://github.com/opensearch-project/data-prepper/blob/f2c9341ad127663b750a19df5f53527e4daf0dc3/data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/AggregateProcessorIT.java#L417-L425

An alternative would be to use `CallableFuture<>`.

Then in either case, put the assertions outside of the background thread.